source/initial-dns-seedlist-discovery/initial-dns-seedlist-discovery.rst

```diff
@@ -97,14 +96,52 @@ It is an error to specify more than one host name in a connection string with
 the ``mongodb+srv`` protocol, and the driver MUST raise a parse error and MUST
 NOT do DNS resolution or contact hosts.
 
-The driver MUST NOT attempt to connect to any hosts until the DNS query has
-returned its results.
+A driver MUST verify that in addition to the ``{hostname}``, the
+``{domainname}`` consists of at least two parts: the domain name, and a TLD.
+Drivers MUST raise an error and MUST NOT contact the DNS server to obtain SRV
+(or TXT records) if the full URI does not consists of at least three parts.
+
+If ``mongodb+srv`` is used, a driver MUST implicitly also enable TLS. Clients
+can turn this off by passing ``ssl=false`` in either the Connection String,
+or options passed in as parameters in code to the MongoClient constructor (or
+equivalent API for each driver), but not through a TXT record (discussed in a
+later section).
+
+
+Querying DNS
+~~~~~~~~~~~~
+
+In this preprocessing step, the driver will query the DNS server for SRV
+records on ``{hostname}.{domainname}``, prefixed with ``_mongodb._tcp.``:
+``_mongodb._tcp.{hostname}.{domainname}``. This DNS query is expected to
+respond with one or more SRV records.
+
+The priority and weight fields in returned SRV records MUST be ignored.
 
 If the DNS result returns no SRV records, or no records at all, or a DNS error
 happens, an error MUST be raised indicating that the URI could not be used to
 find hostnames. The error SHALL include the reason why they could not be
 found.
 
+A driver MUST verify that the host names returned through SRV records have the
+same parent ``{domainname}``. Drivers MUST raise an error and MUST NOT
+initiate a connection to any returned host name which does not share the same
+``{domainname}``.
+
+The driver MUST NOT attempt to connect to any hosts until the DNS query has
+returned its results.
+
+If ``srvMaxHosts`` is zero or greater than or equal to the number of hosts in
+the DNS result, the driver MUST populate the seedlist with all hosts. That is to
+say, the driver now MUST behave the same as if a ``mongodb://`` URI was provided
+with all the host names and port numbers that were returned as part of the DNS
+SRV query result.
```

Comment:
This paragraph will be revised after rebasing. See: https://github.com/mongodb/specifications/pull/1067#discussion_r708624023